src/components/Stepper/Step.tsx

```diff
@@ -0,0 +1,156 @@
+  }
+  &:hover
+    ${/* sc-selector */ BulletCircle},
+    &:focus
+    ${/* sc-selector */ BulletCircle} {
+    fill: ${getColor('blueberry0')};
+  }
+`;
+
+const connectorSize = 2;
+const connectorIndentation = pxToRem(bulletSize / 2 - connectorSize / 2);
+const connectorStyled = css`
+  content: '';
+  display: block;
+  background-color: ${getColor('graphiteHB')};
+`;
+
+const StepConnector = styled.div`
+  flex: 1 1 1rem;
+
+  &::before {
+    ${connectorStyled};
+    height: ${`${connectorSize}px`};
+    margin-top: ${connectorIndentation};
+  }
+`;
+
+const StepContent = styled.div<{ $isLast?: StepProps['isLast'] }>`
+  display: flex;
+  align-items: stretch;
+  min-height: 1rem;
+  margin-top: ${getSpace(SpaceSizes.sm)};
+  margin-bottom: ${getSpace(SpaceSizes.sm)};
+
+  &::before {
+    ${connectorStyled};
+    width: ${`${connectorSize}px`};
+    margin-left: ${connectorIndentation};
+    margin-right: ${connectorIndentation};
+  }
+
+  ${({ $isLast }) =>
+    $isLast &&
+    `
+    &::before {
+      background-color: transparent;
+    }
+  `};
+`;
+
+const Step = forwardRef<HTMLDivElement, React.PropsWithChildren<StepProps>>(
+  (
+    { children, label, summary, index, isLast, shouldShowText, onStepClick },
+    ref,
+  ) => {
+    const { orientation, activeStep, areStepsExpanded } = useContext(
+      StepperContext,
+    );
+    const stepNumber = index + 1;
+    const isActive = activeStep === index;
+    const isDone = activeStep > index;
+    const isPending = activeStep < index;
+
+    const isStepClickable = isDone && isNotUndefined(onStepClick);
+
+    const handleStepClick = () => {
+      onStepClick(index);
+    };
+
+    const stepLabel = (
+      <Inline gap="sm">
+        <StepBullet
+          isActive={isActive}
+          isDone={isDone}
+          isPending={isPending}
+          stepNumber={stepNumber}
+        />
+        {shouldShowText && (
+          <Stack gap={SpaceSizes.xs}>
+            <StepLabel isBold={isActive}>{label}</StepLabel>
+            {isNotUndefined(summary) && (
+              <StepSummary size={TextSizes.md} variant={TextVariants.secondary}>
+                {summary}
+              </StepSummary>
+            )}
+          </Stack>
+        )}
+      </Inline>
+    );
+
+    return (
+      <>
+        <div ref={ref}>
+          {isStepClickable ? (
+            <StepButton onClick={handleStepClick}>{stepLabel}</StepButton>
+          ) : (
+            stepLabel
+          )}
+          {orientation === StepperOrientations.vertical && (
+            <StepContent $isLast={isLast} data-testid="step-content">
+              {(isActive || areStepsExpanded) && (
+                <Padbox paddingSize={SpaceSizes.sm}>{children}</Padbox>
+              )}
+            </StepContent>
+          )}
+        </div>
+        {!isLast && orientation === StepperOrientations.horizontal && (
+          <StepConnector />
+        )}
+      </>
+    );
+  },
+);
+Step.displayName = 'Step';
```

Comment:
👍 nice, we should use `displayName` more in busy components

Reply:
Well display name is usually set for normal React components automatically. In this case the problem was with the test where for some reason wasn't displayName set.
The problem is usually with the styled components that doesn't get the name of the constant